packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -905,6 +909,22 @@ export class ConnectionManager implements IConnectionManager {
             assert(clientSequenceNumber <= this.clientSequenceNumber,
                 0x0f0 /* "Incoming local client seq# > generated by this client" */);
 
+            if (message.type === "noop") {
```

Comment:
Noops can be sent, but never received. Changes affect only receiving part of the story, so no matter how code looks like it can't assert strongly that there are no gaps in clientSequenceNumber's other than noops that were sent but not received. I.e. type of op we receive does not matter. What matters is - when we see a gap, was there a noop sent in this gap or not? If not - that's a violation, as there should be no gaps